cli/src/deno.rs

```diff
@@ -34,14 +34,16 @@ pub async fn run(
     args: Vec<String>,
 ) -> CommandResult {
     let state = ExtensionState::new(api, extension.clone());
-    let phylum_api = Extension::builder("phylum-ext")
-        .middleware(|op| match op.name {
+    let phylum_api = Extension {
+        name: "phylum-ext",
+        middleware_fn: Some(Box::new(|op| match op.name {
             "op_request_permission" => op.disable(),
             _ => op,
-        })
-        .ops(api::api_decls())
-        .state(|deno_state| deno_state.put(state))
-        .build();
+        })),
+        ops: api::api_decls().into(),
+        op_state_fn: Some(Box::new(|deno_state| deno_state.put(state))),
+        ..Default::default()
+    };
```

Comment:
This would perhaps be more stable if we used the `deno_core::extension!` macro... But our `ExtensionState` (and `op_state_fn`) take ownership of the `api` argument of this `run()` function and I couldn't figure out how to make that work with the macro...

Reply:
This is the first time this code had to be changed in an update, so I'm not sure stability is a massive problem? A macro would surely have similar chances of getting changed.

Reply:
I think the macro is less likely to break for a few reasons:

1. The macro is used for all of [deno's own extensions](https://github.com/denoland/deno/tree/main/ext)
2. The macro is the higher level abstraction. So even if `Extension` has breaking changes (like [becoming a trait](https://github.com/denoland/deno_core/blob/24a13fc4290631eb259d11ea1201eef2bce4eca0/core/extensions.rs#L469-L471)), the macro can be updated to account for those changes

But you are correct that this code has remained fairly stable. In fact, this change was only necessary to avoid [a deprecation warning](https://github.com/denoland/deno_core/blob/24a13fc4290631eb259d11ea1201eef2bce4eca0/core/extensions.rs#L472-L474).